### ISSUE TITLE
Update Rule: Valid body lang attribute

### ISSUE THREAD

**Issue by jkodu** [auto-wcag/auto-wcag#348]
Rule: https://auto-wcag.github.io/auto-wcag/rules/SC3-1-2-lang-valid.html

![image](https://user-images.githubusercontent.com/20978252/48265183-c0891e00-e423-11e8-87d0-5ca3817b5d46.png)

Like in the scenario above, where the language text contains only one word, it is not a WCAG failure.

The rule needs to be updated to not flag single words.

**Comment by dd8** [comment]
Word boundaries are hard to identify in languages like Japanese that don't use spaces between words:
https://en.wikipedia.org/wiki/Japanese_punctuation#Space

For example, 日米関係 means 'Japan / US relations'

And the concept of single words doesn't map well to Japanese:

日 means 'sun' 
本 means 'origin'
and '日本' means 'Japan' (sun origin = land of the rising sun)

**Comment by jkodu** [comment]
@ShadowBB - can I get your input on the above comment?

**Comment by ShadowBB** [comment]
@jkodu - I am unsure whether or not these two Chinese characters would constitute a "phrase" or not. I interpreted it as a single word because all the other links in the list where single words but I could be wrong.

The SC states "3.1.2 Language of Parts: The human language of each passage or phrase in the content can be programmatically determined..."

In our interpretation a "phrase" cannot be a single world outside of the context of a sentence structure but is a sequence of 2 or more words that constitute a single "unit" of meaning within a sentence.

**Comment by kasperisager** [comment]
From the Merriam-Webster definition of "phrase":

> a word or group of words forming a syntactic constituent with a single grammatical function

https://www.merriam-webster.com/dictionary/phrase

**Comment by dd8** [comment]
I think problem is the definition of "word" - which varies a lot by language, even in closely related languages:

>Not all languages delimit words expressly. Mandarin Chinese is a very analytic 
>language (with few inflectional affixes), making it unnecessary to delimit words 
>orthographically. However, there are many multiple-morpheme compounds in 
>Mandarin, as well as a variety of bound morphemes that make it difficult to 
>clearly determine what constitutes a word.

https://en.wikipedia.org/wiki/Word#Word_boundaries

>Words differ greatly from language to language. In fact, not all languages
>traditionally have a word for 'word', which means that while words can be
>defined objectively according to certain criteria, they don't necessarily match
>something that is recognised by native speakers.

https://www.quora.com/Do-all-languages-consist-of-words#MoreAnswers

**Comment by jkodu** [comment]
@ShadowBB suggests that this rule be `semi-automated`.

**Comment by Brynanders** [comment]
Rule https://github.com/auto-wcag/auto-wcag/pull/132/files
SC documentation https://www.w3.org/WAI/WCAG21/quickref/#language-of-parts

**Comment by ShadowBB** [comment]
Hey @Brynanders.

You are correct in describing what the rule tests. However because it doesn't test whether or not their should be an attribute at all, it could lead to a false positive. If the attribute is not needed to comply to WCAG (for example, there is less than a "phrase" that is a different language) than it should not fail WCAG if it doesn't contain a valid value.

When the rule would be semi-automated than the applicability could test for invalid values and than the expectation (checked by a human) could check whether or not the attribute was needed in the first place.

A different solution is to incorporate an assumption in the rule stating something along the lines of "The lang attribute is being used to comply to WCAG" similar to what we are now doing with the ARIA rules.

**Comment by annethyme** [comment]
Closing since there is a PR to handle this.
